components/tree/private/item.jsx

```diff
@@ -39,18 +42,92 @@ const handleClick = (event, props) => {
 	}
 };
 
+const findNextNode = (flattenedNodes, node) => {
+	const nodes = flattenedNodes.map(((flattenedNode) => flattenedNode.node));
+	const index = nodes.indexOf(node);
+	return flattenedNodes[(index + 1) % flattenedNodes.length];
+};
+
+const findPreviousNode = (flattenedNodes, node) => {
+	const nodes = flattenedNodes.map(((flattenedNode) => flattenedNode.node));
+	let index = nodes.indexOf(node) - 1;
+	if (index < 0) {
+		index += flattenedNodes.length;
+	}
+	return flattenedNodes[index];
+};
+
+const handleKeyDownDown = (event, props) => {
+	if (props.node.selected) {
+		// Go to the next visible node
+		handleClick(event, props);
+		const flattenedNode = findNextNode(props.flattenedNodes, props.node);
+		props.onClick(event, {
+			node: flattenedNode.node,
+			select: true,
+			treeIndex: flattenedNode.treeIndex,
+		});
+	} else {
+		// Focus is on this node but it's not selected, so select it
+		handleClick(event, props);
+	}
+};
+
+const handleKeyDownUp = (event, props) => {
+	if (props.node.selected) {
+		// Go to the next visible node
+		handleClick(event, props);
+		const flattenedNode = findPreviousNode(props.flattenedNodes, props.node);
+		props.onClick(event, {
+			node: flattenedNode.node,
+			select: true,
+			treeIndex: flattenedNode.treeIndex,
+		});
+	} else {
+		// Focus is on this node but it's not selected, so select it
+		handleClick(event, props);
+	}
+};
+
+const handleKeyDownLeft = (event, props) => {
```

Comment:
`left` should close its branch

Reply:
In a comment below, @arodenbeck implied that this behavior is ok.  (Left arrow selects the parent node.)